src/App.tsx

```diff
@@ -308,19 +338,40 @@ function App() {
         setResult(myResponse);
       }
     } catch (error) {
-      enqueueSnackbar('Failed to retrieve results', { variant: 'error' });
+      enqueueSnackbar('Failed to retrieve results', { variant: 'error', action });
     }
     setLoading(false);
   }
 
+  function login(credential: string | undefined) {
+    setIsLoggedIn(true);
+    const jwt: GoogleJWT = credential ? jwtDecode(credential) : ({} as GoogleJWT);
+    setIDToken(credential);
+    setName(jwt.given_name);
+    setProfilePic(jwt.picture);
+  }
```

Comment:
That'd be a good candidate for a custom hook in the future afaict - so we can subscribe to the global "logged in" status with all components that need it.